transports/tcp/src/lib.rs

```diff
@@ -165,7 +165,7 @@ impl Config {
     pub fn new() -> Self {
         Self {
             ttl: None,
-            nodelay: None,
+            nodelay: Some(true), //Enable Nagle's algorithm by default
```

Comment:
```suggestion
            nodelay: Some(true), // Disable Nagle's algorithm by default
```

Quite the opposite, this will and should disable Nagle's algorithm, i.e. it disables the delaying of TCP packets.

Requires a change to the doc comment of the function as well.